src/lib.rs

```diff
@@ -199,16 +201,13 @@ fn prepare_local_template(args: &Args) -> Result<(TempDir, PathBuf, String), any
     Ok((template_base_dir, template_folder, branch))
 }
 
-fn check_cargo_generate_version(template_config: &Option<Config>) -> Result<(), anyhow::Error> {
-    if let Some(Config {
-        template:
-            Some(config::TemplateConfig {
-                cargo_generate_version: Some(requirement),
-                include: _,
-                exclude: _,
-            }),
-        placeholders: _,
-    }) = template_config
+fn check_cargo_generate_version(template_config: &Config) -> Result<(), anyhow::Error> {
+    if let Some(config::TemplateConfig {
+        cargo_generate_version: Some(requirement),
+        include: _,
+        exclude: _,
+        ignore: _,
```

Comment:
```suggestion
        ..
```
clippy should complain about this I think

Reply:
Unfortunately it didn't, and I didn't think of it either... oh well, good thing you noticed @MarcoIeni. 😃 

I'll look into these minor issues tonight - also need to do some writing for the README. This was just a first attempt at solving the problem, so if y'all like the proposal, I count this a huge success.

My own biggest issue is the use of `liquid` for the conditional expressions. It can be a bit cumbersome, but I don't have a better idea for now. The `truthy` and `falsy` filters were an attempt to ease this issue a little, but it might not be enough - `liquid` is really for rendering text and not for evaluating expressions.

Reply:
I don't know liquid well, but `|truthy` and `|falsy` don't look so bad I think :)
Isn't it possible to use the same syntax as `'{% if crate_type == "bin" %}true{% endif %}'`?
In this way we can also have different placeholders based on string comparisons, not only on boolean expressions.

Reply:
By the way if this requires a longer discussion we could just handle "include files conditionally" in this PR and open a separate issue/PR on "include placeholders conditionally" if that's feasible and if you think it's better.

Reply:
it is indeed possible - its actually the ONLY way if it wasn't for the 2 filters :-)

Reply:
> By the way if this requires a longer discussion we could just handle "include files conditionally" in this PR and open a separate issue/PR on "include placeholders conditionally" if that's feasible and if you think it's better.

I'd actually prefer to just have it all here. The only reason to delay in my mind, is if we'd want something other than `liquid`for the conditional expression syntax - I just don't see why or what.

Regarding the placeholders - I added those as well as the ignore setting, as I really feel like this should all be possible from within `cargo-generate.toml`.

Reply:
> it is indeed possible - its actually the ONLY way if it wasn't for the 2 filters :-)

Oh, so you can use the two filters in addition to that other syntax. I.e. it's just syntax sugar. I see :)